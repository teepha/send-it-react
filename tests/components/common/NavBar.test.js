import React from "react";
import { shallow } from "enzyme";
import sinon from "sinon";
import {
  NavBar,
  mapDispatchToProps
} from "../../../src/components/common/NavBar";

const setUp = () => {
  const props = {
    logout: jest.fn(),
    user: {
      id: 1,
      role: "admin"
    }
  };
  const state = {
    showHambugger: false
  };
  return {
    shallowWrapper: shallow(<NavBar {...props} />),
    state,
    props
  };
};

describe("NavBar Component test", () => {
  it("render navbar without crashing", () => {
    const { shallowWrapper } = setUp();
    expect(shallowWrapper).toMatchSnapshot();
  });

  it("handlelogout should call props.logout", () => {
    const { shallowWrapper } = setUp();
    const handleLogoutSpy = jest.spyOn(
      shallowWrapper.instance(),
      "handleLogout"
    );
    handleLogoutSpy();
    expect(shallowWrapper.instance().props.logout).toBeCalled();
  });

  it("mapdispatch to props should call logout", () => {
    const dispatchSpy = sinon.spy();
    const dispatchResult = mapDispatchToProps(dispatchSpy);

    dispatchResult.logout();
    expect(dispatchSpy.callCount).toEqual(1);
  });

  it("should redirect when role is user role is admin", () => {
    const { shallowWrapper } = setUp();
    shallowWrapper.setProps({
      user: {
        role: "admin"
      }
    });
    expect(shallowWrapper).toMatchSnapshot();
  });

  it("invokes handleNavbarClick method", () => {
    const { shallowWrapper } = setUp();
    const handleNavbarClickSpy = jest.spyOn(
      shallowWrapper.instance(),
      "handleNavbarClick"
    );
    shallowWrapper.instance().handleNavbarClick();
    expect(handleNavbarClickSpy).toHaveBeenCalledWith();
  });
});
