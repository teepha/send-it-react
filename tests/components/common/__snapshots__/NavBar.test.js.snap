// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar Component test render navbar without crashing 1`] = `
<div>
  <header>
    <div
      className="wrapper"
    >
      <div
        id="logo"
      >
        <Link
          className="logo"
          replace={false}
          to="/"
        >
          <img
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </Link>
        <h1
          className="logo"
        >
          <Link
            className="sendIT"
            replace={false}
            to="/"
          >
            SendIT
          </Link>
        </h1>
      </div>
      <nav>
        <ul
          className="navigation "
        >
          <li>
            <Link
              className="navigation-link"
              replace={false}
              to="/admin-profile"
            >
              My Profile
            </Link>
          </li>
          <li>
            <Link
              className="navigation-link"
              onClick={[Function]}
              replace={false}
              to="/"
            >
              Logout
            </Link>
          </li>
        </ul>
      </nav>
      <div
        className="navbar__header"
      >
        <div
          className="navbar__btn "
          onClick={[Function]}
        >
          <div
            className="bar bar1"
          />
          <div
            className="bar bar2"
          />
          <div
            className="bar bar3"
          />
        </div>
      </div>
    </div>
  </header>
</div>
`;

exports[`NavBar Component test should redirect when role is user role is admin 1`] = `
<div>
  <header>
    <div
      className="wrapper"
    >
      <div
        id="logo"
      >
        <Link
          className="logo"
          replace={false}
          to="/"
        >
          <img
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </Link>
        <h1
          className="logo"
        >
          <Link
            className="sendIT"
            replace={false}
            to="/"
          >
            SendIT
          </Link>
        </h1>
      </div>
      <nav>
        <ul
          className="navigation "
        >
          <li>
            <Link
              className="navigation-link"
              replace={false}
              to="/login"
            >
              Login
            </Link>
          </li>
          <li>
            <Link
              className="navigation-link"
              replace={false}
              to="/register"
            >
              SignUp
            </Link>
          </li>
        </ul>
      </nav>
      <div
        className="navbar__header"
      >
        <div
          className="navbar__btn "
          onClick={[Function]}
        >
          <div
            className="bar bar1"
          />
          <div
            className="bar bar2"
          />
          <div
            className="bar bar3"
          />
        </div>
      </div>
    </div>
  </header>
</div>
`;
